File: phalcon/library/tencent/Trtc.php
<?php
namespace Library\Tencent;

use Config\Tencent;
use Library\Curl;
use Util\Util;

/* 实时音视频 */
class Trtc extends Signature {

  static int $SdkAppId = 0;

  /* 公共配置 */
  static function Init() {
    $cfg = Tencent::TRTC();
    self::$ApiUrl = 'https://trtc.tencentcloudapi.com/';
    self::$Host = 'trtc.tencentcloudapi.com';
    self::$Service = 'trtc';
    self::$Version = '2019-07-22';
    self::$SdkAppId = $cfg['SDKAppID'];
  }

  /* 房间-查询 */
  static function RoomList(int $StartTime, int $EndTime) {
    // 参数
    self::Init();
    self::$Action = 'DescribeRoomInformation';
    // 数据
    $time = time();
    $data = [
      'SdkAppId'=> (string)self::$SdkAppId,
      'StartTime'=> $StartTime,
      'EndTime'=> $EndTime,
    ];
    $dataStr = Util::JsonEncode($data);
    // 请求头
    $header = self::V3Header($data);
    return Curl::Request(self::$ApiUrl, $dataStr, 'POST', $header);
  }

  /* 房间-解散 */
  static function RoomDismiss(string $roomId) {
    // 参数
    self::Init();
    self::$Action = 'DismissRoomByStrRoomId';
    // 数据
    $data = [
      'SdkAppId'=> self::$SdkAppId,
      'RoomId'=> $roomId,
    ];
    $dataStr = Util::JsonEncode($data);
    // 请求头
    $header = self::V3Header($data);
    return Curl::Request(self::$ApiUrl, $dataStr, 'POST', $header);
  }

}

File: python/library/tencent/trtc.py
from util.util import Util
from .signature import Signature
from config.tencent import Tencent
from library.curl import Curl

# 实时音视频
class Trtc:

  SdkAppId: int = 0

  # 公共配置
  def Init():
    cfg = Tencent.TRTC()
    Signature.ApiUrl = 'https://trtc.tencentcloudapi.com/'
    Signature.Host = 'trtc.tencentcloudapi.com'
    Signature.Service = 'trtc'
    Signature.Version = '2019-07-22'
    Trtc.SdkAppId = cfg['SDKAppID']

  # 房间-查询
  def RoomList(StartTime: int, EndTime: int):
    # 参数
    Trtc.Init()
    Signature.Action = 'DescribeRoomInformation'
    # 数据
    data = {
      'SdkAppId': str(Trtc.SdkAppId),
      'StartTime': StartTime,
      'EndTime': EndTime,
    }
    # 请求头
    header = Signature.V3Header(data)
    return Curl.Request(Signature.ApiUrl, Util.JsonEncode(data), 'POST', header)

  # 房间-解散
  def RoomDismiss(roomId: str):
    # 参数
    Trtc.Init()
    Signature.Action = 'DismissRoomByStrRoomId'
    # 数据
    time = Util.Time()
    data = {
      'SdkAppId': str(Trtc.SdkAppId),
      'RoomId': roomId,
    }
    # 请求头
    header = Signature.V3Header(data)
    return Curl.Request(Signature.ApiUrl, Util.JsonEncode(data), 'POST', header)